services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala

```diff
@@ -36,85 +41,115 @@ class ProjectConfigurationServiceTest extends AsyncWordSpec with Matchers with M
+      val result: Unit = ()
+      val error = new Exception("Oh no")
+      when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(dummyProject))
+
       "return success if creation was successful" in {
-        val result: Unit = ()
         when(configurationRepository.getById(projectId)).thenReturn(Future.successful(None))
         when(configurationRepository.addConfiguration(activeConfiguration)).thenReturn(Future.successful(result))
-        configurationService.addConfiguration(activeConfiguration).map(_ shouldBe result)
+        configurationService.addConfiguration(activeConfiguration, userId).map(_ shouldBe result)
       }
 
       "return success if update was successful" in {
-        val result: Unit = ()
         when(configurationRepository.getById(projectId)).thenReturn(Future.successful(Some(activeConfiguration)))
         when(configurationRepository.updateConfiguration(updatedConfiguration)).thenReturn(Future.successful(result))
-        configurationService.addConfiguration(updatedConfiguration).map(_ shouldBe result)
+        configurationService.addConfiguration(updatedConfiguration, userId).map(_ shouldBe result)
       }
 
       "return failure if it couldn't fetch existing configuration" in {
-        val error = new Exception("Oh no")
         when(configurationRepository.getById(projectId)).thenReturn(Future.failed(error))
-        configurationService.addConfiguration(activeConfiguration).failed.map(_ shouldBe error)
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
       }
 
       "return failure if creation wasn't successful" in {
-        val error = new Exception("Oh no")
         when(configurationRepository.getById(projectId)).thenReturn(Future.successful(None))
         when(configurationRepository.addConfiguration(activeConfiguration)).thenReturn(Future.failed(error))
-        configurationService.addConfiguration(activeConfiguration).failed.map(_ shouldBe error)
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
       }
 
       "return failure if update wasn't successful" in {
-        val error = new Exception("Oh no")
         when(configurationRepository.getById(projectId)).thenReturn(Future.successful(Some(activeConfiguration)))
         when(configurationRepository.updateConfiguration(activeConfiguration)).thenReturn(Future.failed(error))
-        configurationService.addConfiguration(activeConfiguration).failed.map(_ shouldBe error)
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
+      }
+    }
+
+    "add new configuration for project and user is not project owner" should {
+      val error = new ProjectAccessDeniedException
+      when(projectService.getUserProjectById(projectId, wrongUserId)).thenReturn(Future.failed(error))
+
+      "return failure if user is not the project owner" in {
```

Comment:
```suggestion
      "return failure" in {
```

`ProjectConfigurationService when add new configuration for project and user is not project owner should return failure if user is not the project owner`

Reply:
Done.